scenarios/cpp/windows/captioning/captioning.cpp

```diff
@@ -0,0 +1,351 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
+
+// NOTES:
+// - Install the NuGet package Microsoft.CognitiveServices.Speech.
+// - The Speech SDK on Windows requires the Microsoft Visual C++ Redistributable for Visual Studio 2019 on the system. See:
+// https://docs.microsoft.com/azure/cognitive-services/speech-service/speech-sdk
+// - In Windows, place the following files from the Speech SDK in the same folder as your compiled .exe:
+//     - Microsoft.CognitiveServices.Speech.core.dll
+//     - Microsoft.CognitiveServices.Speech.extension.audio.sys.dll
+
+#include <exception>
+#include <filesystem>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <optional>
+#include <sstream>
+#include <speechapi_cxx.h>
+#include "helper.h"
+
+using namespace Microsoft::CognitiveServices::Speech;
+using namespace Microsoft::CognitiveServices::Speech::Audio;
+using namespace Microsoft::CognitiveServices::Speech::Speaker;
+
+class Captioning
+{
+private:
+
+    std::shared_ptr<UserConfig> m_userConfig = NULL;
+    std::shared_ptr<AudioStreamFormat> m_format = NULL;
+    std::shared_ptr<BinaryFileReader> m_callback = NULL;
+    std::shared_ptr<PullAudioInputStream> m_stream = NULL;
+
+    std::string TimestampFromSpeechRecognitionResult(std::shared_ptr<SpeechRecognitionResult> result)
+    {
+        std::ostringstream strTimestamp;
+        auto timestamp = TimestampFromTicks(result->Offset(), result->Offset() + result->Duration());    
```

Comment:
I know folks have different opinions, but I would never use "auto" in any sample code. I actually don't use it in any code I know someone else will need to work on (which is everything we do at work). Using "auto" makes the code unreadable. We want sample code to be readable.

Reply:
Don't remember if I already commented on this...

Reply:
Travis suggested using "var" when the type can be inferred from the right hand side (he did mention it might be a holy war : )
That said, in this case, the type can't be inferred, so I will replace "auto" here and look for other cases like this.